packages/amplify-nodejs-function-template-provider/src/providers/graphqlRequestProvider.ts

```diff
@@ -0,0 +1,43 @@
+import { FunctionTemplateParameters } from 'amplify-function-plugin-interface';
+import path from 'path';
+import fs from 'fs-extra';
+import {
+  AmplifySupportedService, exitOnNextTick,
+} from 'amplify-cli-core';
+import { getDstMap } from '../utils/destFileMapper';
+import { templateRoot } from '../utils/constants';
+import { $TSContext } from 'amplify-cli-core';
+
+const pathToTemplateFilesIAM = path.join(templateRoot, 'lambda', 'appsync-request');
+
+/**
+ * Graphql request to an AppSync API using Node runtime Lambda function
+ */
+export async function graphqlRequest(context: $TSContext): Promise<FunctionTemplateParameters> {
+  const { allResources } = await context.amplify.getResourceStatus();
+  const apiResource = allResources.filter((resource: { service: string }) => resource.service === AmplifySupportedService.APPSYNC);
+
+  if (apiResource.length === 0) {
+    context.print.error(`${AmplifySupportedService.APPSYNC} API does not exist. To add an api, use "amplify add api".`);
```

Comment:
context.print is deprecated. Please use `printer` from the `amplify-prompts` package.

Reply:
modified to use printer